pkg/csi-common/utils.go

```diff
@@ -102,9 +102,22 @@ func RunControllerandNodePublishServer(endpoint string, d *CSIDriver, cs csi.Con
 	s.Wait()
 }
 
+// requestWithSecrets is an interface for matching gRpc requests that contain secrets that need to be masked
+type requestWithSecrets interface {
+	GetSecrets() map[string]string
+}
+
 func logGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+
+	containsSecrets, origSecrets := maskSecrets(req)
+
 	klog.V(3).Infof("GRPC call: %s", info.FullMethod)
 	klog.V(5).Infof("GRPC request: %+v", req)
+
+	if containsSecrets {
+		unmaskSecrets(req.(requestWithSecrets), origSecrets)
```

Comment:
is it possible to copy the `req` variable, so we don't need to do `unmaskSecrets`, original way will change the `req` value, there is risk for doing this.
Another option is change print way for below statement, that would be safer
```
klog.V(5).Infof("GRPC request: %+v", req)
```

Reply:
the problem with the copy is that the Secrets field will still point to the same map, so wouldn't make any difference - would have to do a deep copy, which I prefer to avoid if possible.
Other options I thought about are string replace, which is quite fragile IMO; and reflection, which is also generally discouraged. So while I agree that mutating the original object is not ideal, it occurred to me as the best viable option. But if you see another one, or you'd prefer either the string replacement or the reflection let me know and I will change it.
thx for the review :)

Reply:
maybe just write a print function to replace `klog.V(5).Infof("GRPC request: %+v", req)`, change the original value is not a good idea, it may have potential data corruption issue.

Reply:
do you mean using a string replace function/regex? for example something like
```
s := fmt.Sprintf("%+v", req)
... replace secrets in s using regex ...
```
I don't think anything else (outside of a copy + reflection) is possible.
copy + reflection:
```
reqCopy := copy // makes a shallow copy of the req
... use reflection to replace the secrets map in reqCopy with a new map with masked secrets ...
```
From those two I like the reflection approach more , is more robust. String replacement/regex is not very reliable imo.

Reply:
`String replacement/regex` and reflection approach are both ok for me, don't change the `req` value is the bottom line IMO. Thanks.